play-json/js/src/main/scala/EnvReads.scala

```diff
@@ -4,8 +4,55 @@
 
 package play.api.libs.json
 
+import scala.util.control
+
 trait EnvReads {
-  // No specific reader
+  /**
+   * Deserializer for BigDecimal
+   */
+  implicit val bigDecReads: Reads[BigDecimal] = Reads[BigDecimal] {
+    case JsString(s) =>
+      control.Exception
+        .catching(classOf[NumberFormatException])
+        .opt(JsSuccess(BigDecimal(new java.math.BigDecimal(s))))
+        .getOrElse(JsError(JsonValidationError("error.expected.numberformatexception")))
+    case JsNumber(d) => JsSuccess(d.underlying)
+    case _           => JsError(JsonValidationError("error.expected.jsnumberorjsstring"))
+  }
+
+  /**
+   * Deserializer for BigDecimal
+   */
+  implicit val javaBigDecReads: Reads[java.math.BigDecimal] = bigDecReads.map(_.underlying)
```

Comment:
Would rather implement it the other way, as implement is mainly based on Java types

Reply:
Implementation based on `java.math.BigDecimal` would mean one extra wrapping in the case when original value is `JsNumber` which uses `scala.math.BigDecimal` underneath. The current implementation introduces extra wrapping when original value is `JsString`. I think I'd better extract only`JsString` case into private def.

Reply:
Current impl is already Java based, just compose the Scala one on it with `map`, not the reverse as now.